tests/omd_tyxml.ml

```diff
@@ -0,0 +1,51 @@
+(* Unit tests for conversion from Omd.t to Tyxml.t *)
+
+let tyxml_to_string t =
+  Format.asprintf "%a" Tyxml.Html.(pp ()) t
+
+let html_of_string s =
+  s |> Omd.of_string |> Omd_tyxml.of_omd |> tyxml_to_string
+
+let test name ~actual ~expected =
+  try
+    assert (actual = expected)
+  with Assert_failure (file, line, _) ->
+    Printf.eprintf "Test '%s' failed (file %s, line %d):\n  expected: %s\n  actual: %s\n\n"
+      name file line expected actual
+
+let () =
+  test "the empty document"
+    ~actual:(html_of_string "")
+    ~expected:{|<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml"><head><title></title></head><body></body></html>|};
+
+  test "a plain line"
+    ~actual:(html_of_string "a plain line")
+    ~expected:{|<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml"><head><title></title></head><body><p>a plain line</p></body></html>|};
+
+  test "emphasized text"
+    ~actual:(html_of_string "some *emphasized* text")
+    ~expected:{||}
```

Comment:
We typically use `dune` `(diff ...)` rules for this kind of tests which gives us `promote` functionality for free.